core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1747,6 +1748,12 @@ object SnappySession extends Logging {
   private[this] val ID = new AtomicInteger(0)
   private[sql] val ExecutionKey = "EXECUTION"
 
+  lazy val isEnterpriseEdition: Boolean = {
+    GemFireVersion.getInstance(Class.forName("io.snappydata.gemxd.SnappyDataVersion"),
```

Comment:
Ok, then remove SnappyDataVersion.isEnterpriseEdition and use SnappySession one throughout.

Also don't quite understand why the SnappyDataVersion class needs to be passed to GemFireVersion.getInstance(). Since this is the regular GemFireXDVersion.properties, the class used should be GemFireXDVersion for which getResourceAsStream should pass fine. Anyway the fact that SnappyDataVersion.properties is created but used only for few things while GemFireXDVersion.properties used for other things makes this whole thing quite messy. Needs a good cleanup later.